app/src/main/java/io/github/droidkaigi/confsched2017/view/fragment/SponsorsFragment.java

```diff
@@ -34,17 +62,104 @@ public void onAttach(Context context) {
+        adapter = new SponsorshipsAdapter(getContext());
+        binding.recyclerView.setAdapter(adapter);
+        binding.recyclerView.addItemDecoration(new DividerItemDecoration(getContext(), DividerItemDecoration.VERTICAL));
+        binding.recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+    }
+
+    private void renderSponsorships(List<SponsorshipViewModel> sponsorships) {
+        adapter.addAllWithNotify(sponsorships);
+    }
+
+    private static class SponsorAdapter extends ArrayRecyclerAdapter<SponsorViewModel, BindingHolder<ViewSponsorCellBinding>>
+            implements SponsorViewModel.Callback {
+
+        public SponsorAdapter(@NonNull Context context) {
+            super(context);
+        }
+
+        @Override
+        public BindingHolder<ViewSponsorCellBinding> onCreateViewHolder(ViewGroup parent, int viewType) {
+            return new BindingHolder<>(getContext(), parent, R.layout.view_sponsor_cell);
+        }
+
+        @Override
+        public void onBindViewHolder(BindingHolder<ViewSponsorCellBinding> holder, int position) {
+            SponsorViewModel viewModel = getItem(position);
+            viewModel.setCallback(this);
+            ViewSponsorCellBinding itemBinding = holder.binding;
+            itemBinding.sponsorLogo.setMinimumHeight(getScreenWidth() / 3);
+            itemBinding.setViewModel(viewModel);
+            itemBinding.executePendingBindings();
+        }
+
+        @Override
+        public void onClickSponsor(String url) {
+            Optional<Intent> intentOptional = IntentHelper.buildActionViewIntent(getContext(), url);
+            intentOptional.ifPresent(intent -> getContext().startActivity(intent));
+        }
+
+        private int getScreenWidth() {
+            Display display = ((SponsorsActivity) getContext()).getWindowManager().getDefaultDisplay();
+            Point size = new Point();
+            display.getSize(size);
+            return size.x;
+        }
+    }
+
+    private static class SponsorshipsAdapter extends
+            ArrayRecyclerAdapter<SponsorshipViewModel, BindingHolder<ViewSponsorshipCellBinding>> {
+
+        public SponsorshipsAdapter(@NonNull Context context) {
+            super(context);
+        }
+
+        @Override
+        public BindingHolder<ViewSponsorshipCellBinding> onCreateViewHolder(ViewGroup parent,
+                int viewType) {
+            return new BindingHolder<>(getContext(), parent, R.layout.view_sponsorship_cell);
+        }
+
+        @Override
+        public void onBindViewHolder(BindingHolder<ViewSponsorshipCellBinding> holder, int position) {
+            SponsorshipViewModel viewModel = getItem(position);
+            ViewSponsorshipCellBinding itemBinding = holder.binding;
+            SponsorAdapter adapter = new SponsorAdapter(holder.itemView.getContext());
+            adapter.addAllWithNotify(viewModel.getSponsors());
+            itemBinding.sponsorhipRecyclerView.setAdapter(adapter);
```

Comment:
Oh GridLayoutManager squishes the images and doesn't scale well. See attached to see what I mean. Let me know if there's a better way to make it look good with GridLayoutManager

<img src="https://cloud.githubusercontent.com/assets/73175/22714591/8f8724dc-edd0-11e6-8b8e-d556fd9f09c4.png" width="300"/>